checks/file_utils.go

```diff
@@ -131,6 +131,61 @@ func CheckFilesContent(shellPathFnPattern string,
 	return res, nil
 }
 
+// FileCbData is any data the caller can act upon
+// to keep state.
+type FileCbData interface{}
+
+// FileContentCb is the callback.
+// The bool returned indicates whether the CheckFilesContent2
+// should continue iterating over files or not.
+type FileContentCb func(path string, content []byte,
+	dl checker.DetailLogger, data FileCbData) (bool, error)
+
+func CheckFilesContent2(shellPathFnPattern string,
+	caseSensitive bool,
+	c *checker.CheckRequest,
+	onFileContent FileContentCb,
+	data FileCbData,
+) error {
+	predicate := func(filepath string) (bool, error) {
+		// Filter out Scorecard's own test files.
+		if isScorecardTestFile(c.Owner, c.Repo, filepath) {
+			return false, nil
+		}
+		// Filter out files based on path/names using the pattern.
+		b, err := isMatchingPath(shellPathFnPattern, filepath, caseSensitive)
+		if err != nil {
+			return false, err
+		}
+		return b, nil
+	}
+
+	matchedFiles, err := c.RepoClient.ListFiles(predicate)
```

Comment:
Would this not burn the api quota like crazy ? When we get the archive, we do it once and then everything is read from that unpacked repo archive. this is like git clone, but one call. now you are doing one per file request, that will choke quota.

Reply:
the repo client is initialized once only, so the tarball is only downloaded once. Not this code is just a copy of the existing `CheckFilesContent()`